README.md

```diff
@@ -1,6 +1,112 @@
+
+   Converts to a int holding a Unix timestamp.
+
+    ```csharp
+        private int? UnixTimestampNullable(object obj)
+        {
+            var date = (DateTime?) obj;
+            return (int?) (date?.Subtract(new DateTime(1970, 1, 1)))?.TotalSeconds;
+        }
+    ```
+
+    **Arrays**:
+
+    ```csharp
+        var fieldValueAsList = (List<ExampleList>) fieldValue;
+        var itemsSchema = GetSchemaForArrayItems(fieldSchema);
+        var recordsList = fieldValueAsList.Select(listItem => PopulateFields(listItem, itemsSchema)).ToArray()
+        record.Add(field.Name, recordsList);
+        
+        private Schema GetSchemaForArrayItems(Schema arraySchema)
+        {
+            var jsonSchema = (JsonElement) JsonSerializer.Deserialize<object>(arraySchema.ToString());
+            jsonSchema.TryGetProperty("items", out var itemsSchemaJson);
+            return Schema.Parse(itemsSchemaJson.ToString());
+        }
+    ``` 
+
+    Once the data types have been converted you can create the Generic Record by adding each value to the record, this can be done through the use of a for loop & reflection:
+
+    ```csharp
+
+        public GenericRecord PopulateFields(object item, Schema schema)
+        {
+            var record = new GenericRecord((RecordSchema) schema);
+            ((RecordSchema) schema).Fields.ForEach(field =>
+            {
+                PropertyInfo propInfo = item.GetType().GetProperty(field.Name);
+                if (propInfo == null)
+                {
+                    Console.WriteLine($"Field name: {field.Name} not found in {item}");
+                    return;
+                }
+
+                var fieldValue = propInfo.GetValue(item);
+                var fieldSchema = field.Schema;
+                var fieldType = field.Schema.Tag;
+
+                //Add any data type conversion required here
+
+                record.Add(field.Name, fieldValue);
+            });
+
+            return record;
+        }
+
+    ```
+
+6. Lastly the record created above is then sent through to Kafka.
+   In order to send the data through to Kafka a SchemaRegistryClient is required. SchemaRegistery are seperate from your Kafka brokers.The Kafka Producers publish the data to Kafka topics and communicates with the Schema Registry to send and recieve schemas that describe the data models for the messages simultaneously. Hence the SchemaRegistry is used to serialize the message and then save the serialize message to Kafka. 
+   We then use the `Producer.Flush(TimeSpan.FromSeconds(10))` to immediately send through the message to Kafka. There is a 10 seconds maximum timeout, this wouldn't hang the listener for 10 seconds every time.Instead, It will wait until either all the messages have been sent or 10 seconds pass. If we remove the 10 seconds then it would wait forever until all the messages have been sent, which we don't want as if there is an issue, it could just hang until the lambda timeout.
```

Comment:
```suggestion
   We then use the `Producer.Flush(TimeSpan.FromSeconds(10))` to immediately send through the message to Kafka. There is a 10 seconds maximum timeout. It will wait until either all the messages have been sent or 10 seconds have passed. If a maximum timeout is not set then it would wait until all the messages have been sent, and in the event there is an issue whilst sending messages, it will likely hang until the lambda timeout which we don't want.
```